src/units/gender.cpp

```diff
@@ -63,11 +63,8 @@ unit_gender::unit_gender(int index, const std::string& name)
 	const unit_gender* gender = from_string(str_gender);
 	return gender ? *gender : fallback;
 }
-/*static*/ const unit_gender* unit_gender::from_int(int index){
-	assert(index < unit_gender::num_genders());
-	auto gender = gender_map[index];
-	assert(gender);
-	return gender;
+/*static*/ gender_list unit_gender::genders(){
+	return {};
```

Comment:
...what? This is just weird, why are there no genders...

Reply:
The `gender_list` class has no members, and only a default constructor. Its function is to provide `begin()` and `end()` methods for iteration. This method simply constructs a `gender_list`.

Reply:
I find this syntax confusing. I'd rather see a constructor call here.